API/models/rbac.go

```diff
@@ -0,0 +1,121 @@
+package models
+
+import (
+	u "p3/utils"
+	"strings"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// Roles
+const (
+	Manager = "manager"
+	User    = "user"
+	Viewer  = "viewer"
+)
+
+// Actions
+const (
+	READ = iota
+	WRITE
+	READONLYNAME
+)
+
+const ROOT_DOMAIN = "*"
+
+func CheckDomainExists(domain string) bool {
+	if domain == ROOT_DOMAIN {
+		return true
+	}
+	x, e := GetEntity(bson.M{"hierarchyName": domain}, "domain", u.RequestFilters{}, nil)
+	return e == "" && x != nil
+}
+
+func RequestGen(x map[string]interface{}, userRoles map[string]string) bool {
```

Comment:
the map could be returned rather than filled

Reply:
Agreed, that's old code, I'll take a better look